lib/srv/db/postgres/test.go

```diff
@@ -420,25 +438,65 @@ func (s *TestServer) handleQueryWithError(client *pgproto3.Backend) error {
 	return nil
 }
 
-func (s *TestServer) handleCreateStoredProcedure(query string) error {
+func (s *TestServer) handleCreateStoredProcedure(query string, pid uint32) error {
 	match := storedProcedureRe.FindStringSubmatch(query)
-	if len(match) != 2 {
+	if match == nil {
 		return trace.BadParameter("failed to extract stored procedure name from query")
 	}
 
-	if _, ok := ephemeralProcs[match[1]]; !ok {
-		if _, ok := procs[match[1]]; !ok {
-			return trace.BadParameter("test server doesn't support stored procedure %q", match[1])
+	if _, ok := procs[match[storedProcedureRe.SubexpIndex("ProcName")]]; !ok {
+		return trace.BadParameter("test server doesn't support stored procedure %q", match[1])
+	}
+
+	procName := storedProcedureName(pid, match[storedProcedureRe.SubexpIndex("Schema")], match[storedProcedureRe.SubexpIndex("ProcName")])
+	var argsCount int
+	args := strings.Split(match[storedProcedureRe.SubexpIndex("Args")], ",")
+	for _, arg := range args {
+		// Skip arguments that have a default value.
+		if !strings.Contains(strings.ToLower(arg), "default") {
+			argsCount++
```

Comment:
On one hand, it's quite impressive to validate things like this, but on the other hand, I feel it's too much custom mock logic. I would prefer something simpler but i don't mind keep this either. 

What's more important is we cover these in our E2E tests. I see both RDS and Redshift auto-user provisioning passed so we are good 👍

Reply:
on that topic, could you check if E2E tests need to be updated (like do we need to adjust the permissions given to the admin user)?